[Tutorial] Added guide to setting up Let's Encrypt SSL certificate on unique port on Hestia Control Panel with Netcup KVM

vdbhb59 opened the pull request:
Added guide to get Let's Encrypt SSL certificate on unique port on Hestia Control Panel.

vdbhb59 committed:
Added guide to get Let's Encrypt SSL certificate on unique port on Hestia Control Panel.

Added guide to get Let's Encrypt SSL certificate on unique port on Hestia Control Panel.

vdbhb59 committed:
Corrected typo

vdbhb59 committed:
Corrected a sentence

vdbhb59 committed:
Few changes

vdbhb59 committed:
Merge branch 'netcup-community:main' into main

vdbhb59 commented:
@stecklars mate, when will you review this? Last you had told me almost 6 or so months ago.

stecklars commented on community-tutorials/hestiacp-add-lets-encrypt-unique-port/01-en.md:
```diff
@@ -0,0 +1,175 @@
+
+- A root server from netcup with latest Ubuntu 20.04/22.04/LTS; Debian 10/11/12/LTS or later installed (see the below URL) - use minimal mode of installation, also called clean installation. [Installation Tutorial](https://github.com/netcup-community/netcup-root-server-hosting-using-hestia-control-panel/01-en.md/)
+- A registered domain name
+- Root (optional) or ssh access to your server
```
Root access and SSH are two different things. We recommend to remove this from the tutorial.

vdbhb59 replied:
Amended to provide proper details.

stecklars commented on community-tutorials/hestiacp-add-lets-encrypt-unique-port/01-en.md:
```diff
@@ -0,0 +1,175 @@
+
+# Step 2: Add the necessary changes to the service (example git)
+
+Login via SSH to the server.
```
This should be the first step, as it is required to install updates on the server.

vdbhb59 replied:
Actually, it is already there in the 1st step. I seem to have duplicated it in error. Removed.

stecklars commented on community-tutorials/hestiacp-add-lets-encrypt-unique-port/01-en.md:
````diff
@@ -0,0 +1,175 @@
+Enter
+```
+
+Then under nginx.conf (**$HESTIADATA\conf\web\git.domain.tld\nginx.conf**) I added
````
The tutorial suggests that a "Hestia" control panel is installed. However, there is no explanation on how to install it and it also isn't part of the prerequisites.

vdbhb59 replied:
It is already as a part of the prerequisites. See line 30. I have given the installation tutorial. Also have added in introduction.

stecklars commented on community-tutorials/hestiacp-add-lets-encrypt-unique-port/01-en.md:
```diff
@@ -0,0 +1,175 @@
+ a2enmod proxy_http
+ a2enmod proxy_balancer
+ a2enmod proxy_wstunnel
+ systemctl restart apache2
```
This can be removed, as the service is restarted in the following right again.

vdbhb59 replied:
I agree, but it is best to keep it there, as many may not realize, and it is best for them to run it again, at least for newbies.

vdbhb59 committed:
Updated as per review

vdbhb59 commented:
@stecklars mate, updated and added comments above.
Ohh, and thanks for taking time out to do the reviews mate. 🙂

stecklars commented:
Dear @vdbhb59,

Thank you for your submission and your patience.

We are currently working on improving our internal processes so that we can review the tutorials dramatically faster in the future. We apologize for the long delay.

I have already reviewed the tutorial a bit and made some change suggestions. That being said, after reading through the whole tutorial multiple times, some questions remain, so here are some ways in which we would prefer to see this improved:

- Provide an example for the situation before and after the tutorial, so that it is clear what the intended goal is here.
- As far as I currently understand it, you are trying to describe a general goal (making a service on a non-standard port in Hestia CP accessible via TLS/SSL) using some specific software (Gitea / Forgejo). However, Gitea is never explicitly mentioned anywhere, which can lead to confusion for the reader. If you want to show the example with Gitea, then it would be better to add some additional more general explanations, so that the reader can apply it to their software, which might be something different from Gitea.

For these reasons, I highly recommend to rework the tutorial in such a way that it is clear, which goal you are trying to reach, and that a new user can better understand it in general, and, if possible, provide more context on Gitea and why it was used here or alternatively, a more general description, applicable to more / alternative services.

Thank you! :)

vdbhb59 commented:
Thanks @stecklars 
I will rework on this tomorrow, as I have one more tutorial to add (if I can get it up by then).
I will revert once I have completed. Thanks for showing patience with me. :)

Commit:
Updated based on review

Commit:
Updated date to current day

Unless @stecklars you state to revert the date?
The current date will make sense tbvh.

Commit:
CR year 25 & one more addition

Final change

vdbhb59 commented:
@stecklars I have updated the guide/tutorial and provided proper background and more information, reasons etc. As per my understanding & being a new user once myself, I think the changes can now be merged. Please let me know. :)

I was not able to get peace of mind to set and update it properly, hence took so much time to revert on this.

Commit:
No hyperlink please

Just realized, why have hyperlink for dummy/sample links?

Commit:
Got what stecklars was speaking about tutorial url

It was github url instead of the community url. I get it now.

Commit:
Updated with latest details

Product now G12 RS2000 (works on all of course)
Hestia details also updated, as few things moved around
Some cosmetics as well.
Please raise a PR, I guess?

vdbhb59 commented:
@stecklars pls can this and the rest be merged? Makes it useless to have spent hours to create a tutorial which goes to waste.
